app/Http/Controllers/OvertimeRequestController.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Toby\Http\Controllers;
+
+use Illuminate\Auth\Access\AuthorizationException;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
+use Inertia\Response;
+use Toby\Actions\OvertimeRequest\AcceptAsTechnicalAction;
+use Toby\Actions\OvertimeRequest\CancelAction;
+use Toby\Actions\OvertimeRequest\CreateAction;
+use Toby\Actions\OvertimeRequest\RejectAction;
+use Toby\Actions\OvertimeRequest\SettleAction;
+use Toby\Domain\OvertimeRequestStatesRetriever;
+use Toby\Enums\SettlementType;
+use Toby\Helpers\YearPeriodRetriever;
+use Toby\Http\Requests\OvertimeRequestRequest;
+use Toby\Http\Resources\OvertimeRequestActivityResource;
+use Toby\Http\Resources\OvertimeRequestResource;
+use Toby\Http\Resources\SimpleUserResource;
+use Toby\Models\OvertimeRequest;
+use Toby\Models\User;
+
+class OvertimeRequestController extends Controller
+{
+    public function index(Request $request, YearPeriodRetriever $yearPeriodRetriever): Response|RedirectResponse
+    {
+        $this->authorize("canUseOvertimeRequestFunctionality", $request->user());
+
+        $status = $request->get("status", "all");
+
+        $overtimeRequests = $request->user()
+            ->overtimeRequests()
+            ->with(["user.permissions", "user.profile"])
+            ->whereBelongsTo($yearPeriodRetriever->selected())
+            ->latest()
+            ->states(OvertimeRequestStatesRetriever::filterByStatusGroup($status, $request->user()))
+            ->paginate();
+
+        $pending = $request->user()
+            ->overtimeRequests()
+            ->whereBelongsTo($yearPeriodRetriever->selected())
+            ->states(OvertimeRequestStatesRetriever::pendingStates())
+            ->count();
+
+        $success = $request->user()
+            ->overtimeRequests()
+            ->whereBelongsTo($yearPeriodRetriever->selected())
+            ->states(OvertimeRequestStatesRetriever::successStates())
+            ->count();
+
+        $failed = $request->user()
+            ->overtimeRequests()
+            ->whereBelongsTo($yearPeriodRetriever->selected())
+            ->states(OvertimeRequestStatesRetriever::failedStates())
+            ->count();
+
+        $settled = $request->user()
+            ->overtimeRequests()
+            ->whereBelongsTo($yearPeriodRetriever->selected())
+            ->states(OvertimeRequestStatesRetriever::settledStates())
+            ->count();
```

Comment:
Maybe we could cache it somehow?

Reply:
added query cache, for vacation requests too